src/neo4j_genai/components/types.py

```diff
@@ -42,44 +42,29 @@ class TextChunks(DataModel):
     chunks: list[TextChunk]
 
 
-class Neo4jProperty(BaseModel):
-    """Represents a Neo4j property.
-
-    Attributes:
-        key (str): The property name.
-        value (Any): The property value.
-    """
-
-    key: str
-    value: Any
-
-
-class Neo4jEmbeddingProperty(BaseModel):
-    """Represents a Neo4j embedding property.
-
-    Attributes:
-        key (str): The property name.
-        value (list[float]): The embedding vector.
-    """
-
-    key: str
-    value: list[float]
-
-
 class Neo4jNode(BaseModel):
     """Represents a Neo4j node.
 
     Attributes:
         id (str): The ID of the node.
         label (str): The label of the node.
-        properties (Optional[list[Neo4jProperty]]): A list of properties associated with the node.
-        embedding_properties (Optional[list[Neo4jEmbeddingProperty]]): A list of embedding properties associated with the node.
+        properties (Optional[dict[str, Any]]): A dictionary of properties attached to the node.
+        embedding_properties (Optional[dict[str, list[float]]]): A list of embedding properties attached to the node.
     """
 
     id: str
     label: str
-    properties: Optional[list[Neo4jProperty]] = None
-    embedding_properties: Optional[list[Neo4jEmbeddingProperty]] = None
+    properties: Optional[dict[str, Any]] = None
+    embedding_properties: Optional[dict[str, list[float]]] = None
+
+    @field_validator("properties", "embedding_properties")
+    @classmethod
+    def check_for_id_properties(
+        cls, v: Optional[dict[str, Any]]
+    ) -> Optional[dict[str, Any]]:
+        if v and "id" in v.keys():
+            raise TypeError("'id' as a property name is not allowed")
```

Comment:
Why this constraint? Could be an issue with the current implementation of the ERExtractor that asks the LLM to assign an unique ID to each extracted entity (and use this ID in the relation extraction).

Reply:
Ok, I got it, `id` is explicit, makes sense.